compiler/qsc_doc_gen/src/generate_docs.rs

```diff
@@ -189,101 +320,103 @@ pub fn generate_docs(
     // Also, items without any metadata (table of content) should come last.
     files.sort_by_key(|file| {
-        (
-            file.1.package.clone(),
-            file.1.namespace.clone(),
-            file.1.name.clone(),
-        )
+        let prefix = if file.0.ends_with("index.md") {
+            "0"
+        } else {
+            "1"
+        };
+        let name_key = format!("{}{}", prefix, file.1.name);
+        (file.1.package.clone(), file.1.namespace.clone(), name_key)
     });
 
     let mut result: Files = files
         .into_iter()
         .map(|(name, metadata, content)| (name, Arc::from(metadata.to_string().as_str()), content))
         .collect();
 
+    // Insert file info for the top index file, index.md.
+    // This file's contents get replaced in the npm/generate_docs.js file.
+    let file_name: Arc<str> = Arc::from("index.md");
+    let file_metadata: Arc<str> = Arc::from("");
+    let file_content: Arc<str> = Arc::from("");
+    result.push((file_name, file_metadata, file_content));
+
     generate_toc(&mut toc, &mut result);
 
     result
 }
 
 fn generate_doc_for_item<'a>(
+    default_package_id: PackageId,
     package: &'a Package,
     package_kind: PackageKind,
     include_internals: bool,
     item: &'a Item,
     display: &'a CodeDisplay,
     files: &mut FilesWithMetadata,
-) -> Option<(Rc<str>, String)> {
-    // Filter items
-    if !include_internals && (item.visibility == Visibility::Internal) {
-        return None;
-    }
-    if matches!(item.kind, ItemKind::Namespace(_, _)) {
-        return None;
-    }
+) -> Option<(Rc<str>, Arc<Metadata>)> {
+    let (true_package, true_item) = resolve_export(
+        default_package_id,
+        package,
+        include_internals,
+        item,
+        display,
+    )?;
 
     // Get namespace for item
     let ns = get_namespace(package, item)?;
 
     // Add file
-    let (metadata, content) = generate_file(package_kind, &ns, item, display)?;
+    let (metadata, content) = if matches!(item.kind, ItemKind::Export(_, _)) {
+        let true_ns = get_namespace(true_package, true_item)?;
+        generate_exported_file_content(
+            package_kind.clone(),
+            &ns,
+            item,
+            display,
+            &true_ns,
+            true_item,
+        )?
+    } else {
+        generate_file_content(package_kind, &ns, item, display)?
+    };
     let file_name: Arc<str> = Arc::from(format!("{ns}/{}.md", metadata.name).as_str());
     let file_content: Arc<str> = Arc::from(content.as_str());
-
-    // Create toc line
-    let line = format!("  - {{name: {}, uid: {}}}", metadata.name, metadata.uid);
-
     let met: Arc<Metadata> = Arc::from(metadata);
-    files.push((file_name, met, file_content));
+    files.push((file_name, met.clone(), file_content));
 
-    // Return (ns, line)
-    Some((ns.clone(), line))
+    // Return (ns, metadata)
```

Comment:
not sure if this comment is particularly practical

Reply:
removed